identity-account-storage/src/storage/memstore.rs

```diff
@@ -177,51 +174,78 @@ impl Storage for MemStore {
 
         Ok(signature)
       }
-      MethodType::X25519KeyAgreementKey2019 => {
+      KeyType::X25519 => {
         return Err(identity_did::Error::InvalidMethodType.into());
       }
-      MethodType::MerkleKeyCollection2021 => {
-        todo!("[MemStore::key_sign] Handle MerkleKeyCollection2021")
-      }
     }
   }
 
-  async fn chain_state(&self, did: &IotaDID) -> Result<Option<ChainState>> {
-    self.chain_states.read().map(|states| states.get(did).cloned())
+  async fn chain_state(&self, account_id: &AccountId) -> Result<Option<ChainState>> {
+    self.chain_states.read().map(|states| states.get(account_id).cloned())
   }
 
-  async fn set_chain_state(&self, did: &IotaDID, chain_state: &ChainState) -> Result<()> {
-    self.chain_states.write()?.insert(did.clone(), chain_state.clone());
+  async fn set_chain_state(&self, account_id: &AccountId, chain_state: &ChainState) -> Result<()> {
+    self.chain_states.write()?.insert(*account_id, chain_state.clone());
 
     Ok(())
   }
 
-  async fn state(&self, did: &IotaDID) -> Result<Option<IdentityState>> {
-    self.states.read().map(|states| states.get(did).cloned())
+  async fn document(&self, account_id: &AccountId) -> Result<Option<IotaDocument>> {
+    self
+      .documents
+      .read()
+      .map(|documents| documents.get(account_id).cloned())
   }
 
-  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> Result<()> {
-    self.states.write()?.insert(did.clone(), state.clone());
+  async fn set_document(&self, account_id: &AccountId, document: &IotaDocument) -> Result<()> {
+    self.documents.write()?.insert(*account_id, document.clone());
 
     Ok(())
   }
 
   async fn purge(&self, did: &IotaDID) -> Result<()> {
-    let _ = self.states.write()?.remove(did);
-    let _ = self.vaults.write()?.remove(did);
-    let _ = self.chain_states.write()?.remove(did);
+    if let Some(account_id) = self.index_get(did).await? {
+      let _ = self.documents.write()?.remove(&account_id);
+      let _ = self.vaults.write()?.remove(&account_id);
+      let _ = self.chain_states.write()?.remove(&account_id);
+
+      self.index.write()?.remove(did);
+    }
+
+    Ok(())
+  }
```

Comment:
It's often useful to know if a command like this did anything, even if it's idempotent. Maybe return a `bool`?